apis/actions.github.com/v1alpha1/version_test.go

```diff
@@ -0,0 +1,60 @@
+package v1alpha1_test
+
+import (
+	"testing"
+
+	"github.com/actions/actions-runner-controller/apis/actions.github.com/v1alpha1"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestIsVersionAllowed(t *testing.T) {
+	t.Parallel()
+	tt := map[string]struct {
+		resourceVersion string
+		buildVersion    string
+		want            bool
+	}{
+		"dev should always be allowed": {
+			resourceVersion: "0.11.0",
+			buildVersion:    "dev",
```

Comment:
We probably need to accept `canary-sha` here as well: https://github.com/actions/actions-runner-controller/blob/a1a8dc5606d7d4e5532994f7c4a245354384d8b3/.github/workflows/global-publish-canary.yaml#L132

Reply:
Yeah, I thought this one was covered by the exact match, but I guess I can add `HasPrefix` on `canary-` to allow mismatches in canary deployments